resolver/resolver.cc

```diff
@@ -1927,40 +1959,65 @@ class ResolveSignaturesWalk {
         for (auto part : parts) {
             if (!current.exists()) {
                 // First iteration
-                if (part != "") {
-                    if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
-                        e.setHeader(
-                            "The string given to `{}` must be an absolute constant reference that starts with `{}`",
-                            method, "::");
+                if (!package) {
+                    if (part != "") {
+                        if (auto e = ctx.beginError(stringLoc, core::errors::Resolver::LazyResolve)) {
+                            e.setHeader(
+                                "The string given to `{}` must be an absolute constant reference that starts with `{}`",
+                                method, "::");
+                        }
+                        return;
+                    }
+                    current = core::Symbols::root();
+                    continue;
+                } else if (package) {
```

Comment:
Why `else  if (package)` here? It looks like this is the else branch of the conditional on line `1962`, which is already testing `!package`.